src/wayland/dmabuf/mod.rs

```diff
@@ -617,7 +621,7 @@ impl DmabufState {
     pub fn create_global_with_filter<D, F>(
         &mut self,
         display: &DisplayHandle,
-        formats: Vec<Format>,
+        formats: impl IntoIterator<Item = Format>,
```

Comment:
So why do we switch this constructor to use an Iterator and not the other we are internally calling?

Reply:
The whole change is not really needed for this, I just thought it might make the API a bit nicer and more consistent with the other functions. The reason for not changing the internal function is that it uses an `Option` and `Option<impl IntoIterator<Item = Format>>` is a pain to call (there are ways around that, but I did not feel like there is a need for a nicer solution)